test/TestNestedJinjaRule.py

```diff
@@ -0,0 +1,209 @@
+      set_fact:
+        var_three_wld: "{{ {{ {{ {{ {{ 234 }} }} }} }} }}"
+''')
+
+FAIL_TASK_W_5LN_M = PlayFile('playbook.yml', '''
+- hosts: all
+  tasks:
+    - name: five-level wild multiline nesting
+      set_fact:
+        var_three_wld_ml: >
+          {{
+          {{
+          {{
+          {{
+          {{ 234 }}
+          }}
+          }}
+          }}
+          }}
+''')
+
+SUCCESS_TASK_P = PlayFile('playbook.yml', '''
+- hosts: all
+  tasks:
+    - name: non-nested example
+      set_fact:
+        var_one: "number for 'one' is {{ 2 * 1 }}"
+''')
+
+SUCCESS_TASK_P_M = PlayFile('playbook.yml', '''
+- hosts: all
+  tasks:
+    - name: multiline non-nested example
+      set_fact:
+        var_one_ml: >
+          number for 'one' is {{
+          2 * 1 }}
+''')
+
+SUCCESS_TASK_2P = PlayFile('playbook.yml', '''
+- hosts: all
+  tasks:
+    - name: nesting far from each other
+      set_fact:
+        var_two: "number for 'two' is {{ 2 * 1 }} and number for 'three' is {{ 4 - 1 }}"
+''')
+
+SUCCESS_TASK_2P_M = PlayFile('playbook.yml', '''
+- hosts: all
+  tasks:
+    - name: multiline nesting far from each other
+      set_fact:
+        var_two_ml: >
+          number for 'two' is {{ 2 * 1
+          }} and number for 'three' is {{
+          4 - 1 }}
+''')
+
+SUCCESS_TASK_C_2P = PlayFile('playbook.yml', '''
+- hosts: all
+  tasks:
+    - name: nesting close to each other
+      set_fact:
+        var_three: "number for 'ten' is {{ 2 - 1 }}{{ 3 - 3 }}"
+''')
+
+SUCCESS_TASK_C_2P_M = PlayFile('playbook.yml', '''
+- hosts: all
+  tasks:
+    - name: multiline nesting close to each other
+      set_fact:
+        var_three_ml: >
+          number for 'ten' is {{
+          2 - 1
+          }}{{ 3 - 3 }}
+''')
+
+
+@pytest.fixture
+def runner(tmp_path, default_rules_collection):
+    return Runner(
+        default_rules_collection,
+        str(tmp_path / 'playbook.yml'),
+        [], [], [],
+    )
+
+
+@pytest.fixture
+def _playbook_file(tmp_path, request):
+    if request.param is None:
+        return
+    for play_file in request.param:
+        p = tmp_path / play_file.name
+        p.write_text(play_file.content)
+
+
+@pytest.mark.parametrize(
+    '_playbook_file',
+    (
+        pytest.param([FAIL_TASK_1LN], id='file includes one-level nesting'),
+        pytest.param([FAIL_TASK_1LN_M], id='file includes one-level multiline nesting'),
+        pytest.param([FAIL_TASK_2LN], id='file includes two-level nesting'),
+        pytest.param([FAIL_TASK_2LN_M], id='file includes two-level multiline nesting'),
+        pytest.param([FAIL_TASK_W_5LN], id='file includes five-level wild nesting'),
+        pytest.param([FAIL_TASK_W_5LN_M], id='file includes five-level wild multiline nesting'),
+    ),
+    indirect=['_playbook_file'],
+)
+@pytest.mark.usefixtures('_playbook_file')
+def test_including_wrong_nested_jinja(runner):
+    rule_violations = runner.run()
+    assert rule_violations[0].rule.id == '207'
```

Comment:
On the second thought, it's even better to just check if the rule is an instance of our target.
```suggestion
    assert type(rule_violations[0].rule) is NestedJinjaRule  # avoid matching subclasses
```